src/pages/index.js

```diff
@@ -0,0 +1,382 @@
+  console.log(evt.key);
+  if (evt.key === "Escape") {
+    const activeModal = document.querySelector(".modal_opened");
+    console.log(activeModal);
+    closeModal(activeModal);
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+
+  document.addEventListener("keyup", handleEscape);
+  modal.addEventListener("click", handleOverlayClick);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keyup", handleEscape);
+  modal.removeEventListener("click", handleOverlayClick);
+}
+
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  console.log("handleEditFormSubmit triggered!");
+
+  // Change button text to "Saving..."
+  const profileSubmitButton = evt.submitter;
+
+  setButtonText(profileSubmitButton, true);
+
+  // Make API call to edit user info
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      console.log("Profile updated successfully!", data);
+
+      // Update the profile name and description on the page
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+
+      // Close the modal
+      closeModal(editModal);
+    })
+    .catch((err) => {
+      console.error("Error editing profile", err);
+    })
+    .finally(() => {
+      // Reset button text after API call completes
+      setButtonText(profileSubmitButton, false);
+    });
+}
+
+disableButton(cardSubmitButton, settings);
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  console.log(descriptionInput.value);
+  console.log(cardLinkInput.value);
+
+  const inputValues = {
+    name: descriptionInput.value,
+    link: cardLinkInput.value,
+  };
+
+  const cardModalButton = evt.submitter; // Ensure correct button reference
+
+  // Change button text to "Saving..."
+  setButtonText(cardModalButton, true);
+
+  api
+    .addNewCard(inputValues)
+    .then((data) => {
+      renderCard(data, "prepend"); // Use API response instead of inputValues
+
+      // Reset form inputs
+      evt.target.reset();
+
+      // Disable submit button after submission
+      disableButton(cardSubmitButton, settings);
+
+      // Close the modal
+      closeModal(cardModal);
+    })
+    .catch((err) => {
+      console.error("Error adding card:", err);
+    })
+    .finally(() => {
+      // Reset button text after API call completes
+      setButtonText(cardModalButton, false);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault(); // Prevent default form submission behavior
+
+  console.log("avatarInput.value", avatarInput.value);
+  setButtonText(avatarSubmitButton, true);
+
+  api
+    .editAvatarInfo(avatarInput.value) // wrapping the object
+    .then((data) => {
+      console.log("data.avatar", data.avatar);
+      avatarImage.src = data.avatar;
+      setButtonText(avatarSubmitButton, false);
```

Comment:
`      setButtonText(avatarSubmitButton, false);
`

It should be in `.finally`

You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails.